clarity/tests/arkadiko-dao_test.ts

```diff
@@ -0,0 +1,41 @@
+import {
+  Account,
+  Chain,
+  Clarinet,
+  Tx,
+  types,
+} from "https://deno.land/x/clarinet@v0.6.0/index.ts";
+
+Clarinet.test({
+  name: "DAO: add new contract",
+  async fn(chain: Chain, accounts: Map<string, Account>) {
+    let deployer = accounts.get("deployer")!;
+
+    let block = chain.mineBlock([
+      Tx.contractCall("arkadiko-dao", "add-contract-address", [
+        types.ascii("amazing-stacker-that-lends"),
+        types.principal("STSTW15D618BSZQB85R058DS46THH86YQQY6XCB7"),
+        types.principal("STSTW15D618BSZQB85R058DS46THH86YQQY6XCB7.arkadiko-mock-stacker-v1-1")
+      ], deployer.address)
+    ]);
+
+    block.receipts[0].result.expectOk().expectBool(true);
+  }
+});
+
+Clarinet.test({
```

Comment:
Q: I wrongly assumed the current set-contract-address could overwrite contracts. If we deploy an update of the staking contracts, the old ones need to be removed from the DAO. Otherwise both old and new contracts will be able to mint tokens. We could now test this with the mock-stacker.

Reply:
@nieldeckx The current `set-contract-address` can overwrite map values (not specific smart contracts?). Look at the difference of `map-set` vs `map-insert`. Or am I missing something?

Reply:
Ah yes, you are right.